src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/QueueHealthDecider.java

```diff
@@ -0,0 +1,125 @@
+
+import java.util.ArrayList;
+import java.util.List;
+
+// This is a sample decider implementation to finalize decision maker interfaces.
+// TODO: 1. Read action priorities from a configurable yml
+// TODO: 2. Read current queue capacity from NodeConfigurationRca (PR #252)
+
+public class QueueHealthDecider extends Decider {
+
+  public static final String NAME = "queue_health";
+
+  private QueueRejectionClusterRca queueRejectionRca;
+  List<String> actionsByUserPriority = new ArrayList<>();
+  private int counter = 0;
+
+  public QueueHealthDecider(long evalIntervalSeconds, int decisionFrequency, QueueRejectionClusterRca queueRejectionClusterRca) {
+    // TODO: Also consume NodeConfigurationRca
+    super(evalIntervalSeconds, decisionFrequency);
+    this.queueRejectionRca = queueRejectionClusterRca;
+    configureActionPriority();
+  }
+
+  @Override
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public Decision operate() {
+    Decision decision = new Decision(System.currentTimeMillis(), NAME);
+    counter += 1;
+    if (counter < decisionFrequency) {
+      return decision;
+    }
+
+    counter = 0;
+    if (queueRejectionRca.getFlowUnits().isEmpty()) {
+      return decision;
+    }
+
+    HotClusterSummary clusterSummary = queueRejectionRca.getFlowUnits().get(0).getSummary();
+    for (HotNodeSummary nodeSummary : clusterSummary.getHotNodeSummaryList()) {
+      NodeKey esNode = new NodeKey(nodeSummary.getNodeID(), nodeSummary.getHostAddress());
+      for (HotResourceSummary resource : nodeSummary.getHotResourceSummaryList()) {
+        decision.addAction(computeBestAction(esNode, resource.getResource().getResourceEnum()));
+      }
+    }
+
+    return decision;
+  }
+
+  private void configureActionPriority() {
+    // TODO: Input from user configured yml
+    this.actionsByUserPriority.add(ModifyQueueCapacityAction.NAME);
+  }
+
+  /**
+   * Evaluate the most relevant action for a node
+   *
+   * <p>Action relevance decided based on user configured priorities for now, this can be modified
+   * to consume better signals going forward.
+   */
+  private Action computeBestAction(NodeKey esNode, ResourceEnum threadPool) {
+    Action action = null;
+    for (String actionName : actionsByUserPriority) {
+      action =
+          getAction(actionName, esNode, threadPool, getNodeQueueCapacity(esNode, threadPool), true);
+      if (action != null) {
+        break;
+      }
+    }
+    return action;
+  }
+
+  private Action getAction(String actionName, NodeKey esNode, ResourceEnum threadPool, int currCapacity, boolean increase) {
+    switch (actionName) {
+      case ModifyQueueCapacityAction.NAME:
+        return configureQueueCapacity(esNode, threadPool, currCapacity, increase);
+      default:
+        return null;
+    }
+  }
+
+  private ModifyQueueCapacityAction configureQueueCapacity(NodeKey esNode, ResourceEnum threadPool, int currentCapacity, boolean increase) {
```

Comment:
Because this method can return `null`, do we want to make it explicit using the annotation `@nullable` ?